cmd/thanos/query.go

```diff
@@ -347,6 +429,40 @@ type gossipSpec struct {
 	peer *cluster.Peer
 }
 
+type fileSDCache struct {
+	tgs map[string]*targetgroup.Group
+	sync.Mutex
+}
+
+func newFileSDCache() *fileSDCache {
+	return &fileSDCache{
+		tgs: make(map[string]*targetgroup.Group),
+	}
+}
+
+func (f *fileSDCache) update(tgs []*targetgroup.Group) {
+	f.Lock()
+	defer f.Unlock()
+	for _, tg := range tgs {
+		// Some Discoverers send nil target group so need to check for it to avoid panics.
+		if tg != nil {
```

Comment:
filter out wrong cases to avoid indents and have cleaner code.

Reply:
Do you mean something like this:
```
if tg == nil {
    continue
}
f.tgs[tg.Source] = tg
```